### ISSUE TITLE
Recursive option

### ISSUE THREAD

**Issue by mlaferrera** [giannitedesco/minotaur#3]
Thanks for sharing this. I will definitely be leveraging this in the future. Do you have any plans to add support for recursive notifications as well? I have a few use cases where this would be extremely useful[1].

[1] https://github.com/PUNCH-Cyber/stoq-plugins-public/blob/master/dirmon/dirmon/dirmon.py#L46-L68

**Comment by giannitedesco** [comment]
Yes, it's planned for the higher level `Minotaur` object, at the moment it's very vague, I haven't honed in on what I think would be the long-term stable interface for that. Should there just be a recursive flag for `add_watch()` perhaps?

**Comment by mlaferrera** [comment]
I think you are spot on. Calling `add_watch(recursive=True)` would be the simplest way from a user perspective. This is also similar to how `watchdog`[1] handles recursive notifications which has worked very well for me in the past.

[1] https://pypi.org/project/watchdog/

**Comment by giannitedesco** [comment]
Yeah, I think at that point it must make sure `CREATE|DELETE|RENAME_FROM|RENAME_TO` are added in to the mask, then it needs to add the watch, then do a `scandir()` to recurse through pre-existing directory structure to add new watches, and then finally, for each event, to make sure it updates its internal representation of the watches whenever one gets renamed or removed.

**Comment by roma-glushko** [comment]
Oh man, I have reviewed up to 5 projects that provides asyncio interface around inotify and none of them support the recursive option 😌
